client/src/Pages/Profile/UserInfo/index.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import {
+  Typography,
+  Avatar,
+  Grid,
+  Paper,
+  CircularProgress,
+} from '@material-ui/core';
+import { Alert } from '@material-ui/lab';
+
+import Button from '../../../Components/Button';
+
+import useStyles from './style';
+
+function UserInfo({ getUserName }) {
+  const classes = useStyles();
+  const [user, setUser] = useState({});
+  const [loading, setLoading] = useState('');
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const fetchingData = async (isCurrent) => {
+    try {
+      setLoading(true);
+      const { data } = await axios.get(`api/v1/users`);
+      if (isCurrent) {
+        setUser(data.data);
+        getUserName(data.data.username);
+        setLoading(false);
+      }
+    } catch (error) {
+      setLoading(false);
```

Comment:
`setLoading` and `setErrorMessage` will also fail if the component is unmounted and `isCurrent` `is false`